File: CabinRent/CabinRent.Services/Database/Rezervacija.cs
using System;
using System.Collections.Generic;

namespace CabinRent.Services.Database
{
    public partial class Rezervacija
    {
        public Rezervacija()
        {
            DetaljiRezervacijes = new HashSet<DetaljiRezervacije>();
        }

        public int RezervacijaId { get; set; }
        public DateTime DatumPrijave { get; set; }
        public DateTime DatumOdjave { get; set; }
        public int BrojOdraslih { get; set; }
        public int BrojDjece { get; set; }
        public bool Otkazano { get; set; }
        public int KlijentId { get; set; }
        public int ObjekatId { get; set; }

        public virtual Klijent Klijent { get; set; } = null!;
        public virtual Objekat Objekat { get; set; } = null!;
        public virtual ICollection<DetaljiRezervacije> DetaljiRezervacijes { get; set; }
    }
}

File: CabinRent/UI/cabinrent_admin/lib/models/rezervacija.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija{
int? rezervacijaId;
String? datumPrijave;
String? datumOdjave;
int? brojOdraslih;
int? brojDjece;
bool otkazano;
int? objekatId;
Rezervacija(this.rezervacijaId,this.datumPrijave,this.datumOdjave, this.brojOdraslih,
this.brojDjece,this.otkazano,this.objekatId);

factory Rezervacija.fromJson(Map<String,dynamic> json)=>_$RezervacijaFromJson(json);

Map<String,dynamic> toJson()=>_$RezervacijaToJson(this);
}
